custom_components/nest_protect/coordinator.py

```diff
@@ -31,18 +31,37 @@ def __init__(
             update_interval=update_interval,
         )
 
-        self.api = client
+        self.client = client
         self.platforms = []
         self.areas = areas
         self.devices: dict[str, Bucket] = {b.object_key: b for b in devices}
 
     async def _async_update_data(self):
         """Update data."""
         try:
-            # TODO Subscribe to Google Nest pubsub event / subscribe endpoint
 
-            # TODO Pull new buckets and return this here
-            # TODO Needs change in entity.py
             return self.devices
+
+            # access_token = await self.client.get_access_token()
+            # nest = await self.client.authenticate(access_token)
+
+            # transport_url = self.first_data["service_urls"]["urls"]["transport_url"]
+
+            # # TODO Subscribe to Google Nest pubsub event / subscribe endpoint
+            # # TODO Pull new buckets and return this here
+            # print("get_data")
+            # result = await self.client.subscribe_for_data(
```

Comment:
This endpoint (added to lib in same PR) delivers new states. We need to do a post (with a long timeout) and when a new event happens, this will return all new states.

If we receive new states, or if the connection times out, we need to call the POST endpoint again etc. etc.